social-network-api/src/main/java/fs/socialnetworkapi/controller/LikeController.java

```diff
@@ -0,0 +1,43 @@
+package fs.socialnetworkapi.controller;
+
+import fs.socialnetworkapi.dto.post.PostDtoOut;
+import fs.socialnetworkapi.service.LikeService;
+import fs.socialnetworkapi.service.PostService;
+import lombok.AllArgsConstructor;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+@AllArgsConstructor
+@RequestMapping("/api/v1/likes")
+public class LikeController {
+  private final LikeService likeService;
+  private final PostService postService;
+
+  @GetMapping("/post/{postId}")
+  public PostDtoOut getLikesForPost(@PathVariable Long postId) {
+    return postService.findById(postId);
+  }
+
+  @GetMapping("/posts")
+  public List<PostDtoOut> getLikesForPosts(@RequestParam List<Long> postIds) {
+    return postService.findByIds(postIds);
+  }
```

Comment:
You don't need to create new endpoints for posts here. We already have PostController which returns posts. You can leave them as is, but add likes to the PostDtoOut. In the LikesController we need only one endpoint for the like/dislike action.

Reply:
I left it here just in case if it will be useful in future. Likes are already injected as a field in PostDtoOut. So no problem to delete these methods if you say so.

Reply:
It won't be useful in future. Please remove :)